ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardChain.java

```diff
@@ -118,9 +173,9 @@ public synchronized void dropFirstHeader() {
     headers.drop(firstStoredAncestor.get().getHash());
     final Optional<Hash> hash = chainStorage.get(firstStoredAncestor.get().getHash());
     chainStorage.drop(firstStoredAncestor.get().getHash());
-    firstStoredAncestor = hash.flatMap(headers::get);
+    updateFirstStoredAncestor(hash.flatMap(headers::get));
     if (firstStoredAncestor.isEmpty()) {
-      lastStoredPivot = Optional.empty();
+      updateLastStorePivot(Optional.empty());
```

Comment:
Don't think this is covered by a test? I can comment out whole condition and nothing breaks

Reply:
probably there is a lot yet to cover 😅